### PR TITLE
Removing defensive copies

### PR DESCRIPTION
**Pull request by michalczerwinski**
This is removing some defencive ToList() which are not needed: .NET Core 3.0 added the possibility to safely remove entries from Dictionary while enumerating (dotnet/coreclr#18854).

This is for #18598 
No new tests were added as all modified cases are already covered.

### CODE REVIEW

**Comment by ErikEJ** [comment]
Assume you mean ToList, not ToString ?

**Comment by michalczerwinski** [comment]
> Assume you mean ToList, not ToString ?
Yes, my bad. ToList()

File: src/EFCore.Relational/Metadata/Conventions/TableSharingConcurrencyTokenConvention.cs [hunk 1]
```diff
@@ -236,7 +236,7 @@ private static void RemoveDerivedEntityTypes<T>(Dictionary<IConventionEntityType
         {
             var toRemove = new HashSet<KeyValuePair<IConventionEntityType, T>>();
```
Comment (AndriySvyryd) [inline src/EFCore.Relational/Metadata/Conventions/TableSharingConcurrencyTokenConvention.cs:237]: Remove this variable

Comment (michalczerwinski) [reply src/EFCore.Relational/Metadata/Conventions/TableSharingConcurrencyTokenConvention.cs]: You are right, removing.

**Review by AndriySvyryd — APPROVED** [review]

**Comment by AndriySvyryd** [comment]
Thanks for your contribution!

**Pull request merged**